super-ddf.c

```diff
@@ -2971,7 +2977,8 @@ static int __write_ddf_structure(struct dl *d, struct ddf_super *ddf, __u8 type)
 	header->openflag = 1;
 	header->crc = calc_crc(header, 512);
 
-	lseek64(fd, sector<<9, 0);
+	if (lseek64(fd, sector<<9, 0) < 0)
+		goto out;
```

Comment:
the documentation specifies that in case of an error, lssek returns "(off_t) -1" value, so please change all checks to "== -1L" or add a new define for the expected error value e.g #define LSEEK_ERROR ((off_t) -1)